packages/n3h-mock/lib/index.js

```diff
@@ -261,22 +426,92 @@ class N3hMock extends AsyncClass {
     return this._memory[dnaAddress]
   }
 
-  _CatDnaAgent (DnaHash, AgentId) {
-    return DnaHash + '::' + AgentId
+  _checkRequest (requestId) {
+    if (!(requestId in this._requestBook)) {
+      return ''
+    }
+    let bucketId = this._requestBook[requestId]
+    delete this._requestBook[requestId]
+    return bucketId
   }
 
   _track (dnaAddress, agentId, fromZmqId) {
     const ref = this._getMemRef(dnaAddress)
+    // store agent (this will map agentId to transportId)
     ref.mem.insert({
       type: 'agent',
       dnaAddress: dnaAddress,
       agentId: agentId,
       transportId: fromZmqId
     })
-
-    const uid = this._CatDnaAgent(dnaAddress, agentId)
-    log.t("tracking: '" + uid + "' for " + fromZmqId)
+    // also map agentId to transportId with in _senders
+    // const bucketId = this._CatDnaAgent(dnaAddress, agentId)
+    // log.t("tracking: '" + bucketId + "' for " + fromZmqId)
     this._senders[agentId] = fromZmqId
+
+    // send get lists
+    let requestId = this._createRequest(dnaAddress, agentId)
+    this._ipc.send('json', {
+      method: 'handleGetPublishingEntryList',
+      dnaAddress,
+      _id: requestId
+    })
+    requestId = this._createRequest(dnaAddress, agentId)
+    this._ipc.send('json', {
+      method: 'handleGetHoldingEntryList',
+      dnaAddress,
+      _id: requestId
+    })
+    requestId = this._createRequest(dnaAddress, agentId)
+    this._ipc.send('json', {
+      method: 'handleGetPublishingMetaList',
+      dnaAddress,
+      _id: requestId
+    })
+    requestId = this._createRequest(dnaAddress, agentId)
+    this._ipc.send('json', {
+      method: 'handleGetHoldingMetaList',
+      dnaAddress,
+      _id: requestId
+    })
+  }
+
+  _CatDnaAgent (DnaHash, AgentId) {
+    return '' + DnaHash + '::' + AgentId
+  }
+
+  _generateRequestId () {
+    this._requestCount += 1
+    return 'req_' + this._requestCount
+  }
+
+  // create and return a new request_id
+  _createRequest (dnaAddress, agentId) {
+    let bucketId = this._CatDnaAgent(dnaAddress, agentId)
+    return this._createRequestWithBucket(bucketId)
+  }
+
+  _createRequestWithBucket (bucketId) {
+    let reqId = this._generateRequestId()
+    this._requestBook[reqId] = bucketId
+    return reqId
+  }
+
+  _bookkeepAddress (book, dnaAddress, agentId, address) {
+    const bucketId = this._CatDnaAgent(dnaAddress, agentId)
+    this._bookkeepAddressWithBucket(book, bucketId, address)
+  }
+
+  _bookkeepAddressWithBucket (book, bucketId, address) {
+    if (!(bucketId in book)) {
+      let array = []
+      array.push(address)
+      book[bucketId] = array
+      return
+    }
+    let array = book[bucketId]
+    array.push(address)
+    book[bucketId] = array
```

Comment:
see foreshortening in hackmode version of this fn

Reply:
done